common/src/main/java/com/wynntils/screens/maps/SeaskipperMapScreen.java

```diff
@@ -0,0 +1,522 @@
+                            TextShadow.OUTLINE);
+
+            renderYOffset += 10;
+
+            FontRenderer.getInstance()
+                    .renderText(
+                            poseStack,
+                            "Click to go here!",
+                            10,
+                            10 + renderYOffset,
+                            CommonColors.BLUE,
+                            HorizontalAlignment.LEFT,
+                            VerticalAlignment.TOP,
+                            TextShadow.OUTLINE);
+        } else if (origin) {
+            renderYOffset += 30;
+
+            FontRenderer.getInstance()
+                    .renderText(
+                            poseStack,
+                            "Origin",
+                            10,
+                            10 + renderYOffset,
+                            CommonColors.RED,
+                            HorizontalAlignment.LEFT,
+                            VerticalAlignment.TOP,
+                            TextShadow.OUTLINE);
+        } else {
+            renderYOffset += 30;
+
+            FontRenderer.getInstance()
+                    .renderText(
+                            poseStack,
+                            "Inaccessible",
+                            10,
+                            10 + renderYOffset,
+                            CommonColors.GRAY,
+                            HorizontalAlignment.LEFT,
+                            VerticalAlignment.TOP,
+                            TextShadow.OUTLINE);
+        }
+
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        seaskipperPoi.getName(),
+                        7,
+                        textureWidth,
+                        Texture.TERRITORY_TOOLTIP_TOP.height() + centerHeight,
+                        Texture.TERRITORY_TOOLTIP_TOP.height() + centerHeight + Texture.TERRITORY_NAME_BOX.height(),
+                        0,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.LEFT,
+                        VerticalAlignment.MIDDLE,
+                        TextShadow.OUTLINE);
+
+        poseStack.popPose();
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+        for (GuiEventListener child : children()) {
+            if (child.isMouseOver(mouseX, mouseY)) {
+                child.mouseClicked(mouseX, mouseY, button);
+                return true;
+            }
+        }
+
+        for (Poi poi : seaskipperPois) {
+            if (poi instanceof SeaskipperPoi destination) {
+                if (destination.isSelected(mouseX, mouseY)) {
+                    buyPass(destination.getName());
+
+                    return true;
+                }
+            }
+        }
+
+        return super.mouseClicked(mouseX, mouseY, button);
+    }
+
+    // Potentially a better way to do this
```

Comment:
There is a way. Use territories, you can match the names to territories, and get their boundaries. But I am pretty sure Wynn just added some off map place for old events or whatnot? I don't see that in this list? It might be only accessible from an island.. Either way. That obviously won't be a territory, so you will need "manual storage" too. I think SeaskipperModel is a perfect place to generate these for you (and store the special cases).

Reply:
Yeah I was going to use territories but the boundaries for some aren't great, like the volcanic isles are split in 2 down the middle and also the way it figures out which port you are currently at needs the boundary to contain the seaskipper location and not all of them do. Can try looking into the manual storage tho as don't really need to generate them each time you open the map

Reply:
In that case, please make it a json in https://github.com/Wynntils/WynntilsWebsite-API/, update urls.json to include it, both on the API repo and in the mod (also increment the version). Then you can fetch it in the model.

Reply:
Ah looks like there's already one there. I presume that's how legacy did it as I tried to not look at the legacy code this time, only needed it for figuring out how to register clicks on the ports.

It still needs updating anyway as it doesn't have that new event island on it which I forgot to include in this too. Also, I'm assuming we get the pois for the map from the Wynn API, that's why there's no seaskipper poi on the map there

Reply:
Artemis uses a community telemetry collected poi set + handcrafted data for pois

Reply:
Also, please just make it a new file. We don't want to share a file with legacy, in case it needs updating. This way you can also change the format, if needed.

Reply:
Before I make this, is there a specific name I should give the file, or is something like seaskipperArtemis fine?

Reply:
`seaskipper_locations.json` is fine, it is different from legacy.

Reply:
I think I've done this correctly, not 100% sure. Only thing I know is wrong is that I don't have the md5 for the json file, how do I find that out?